rust/main/chains/hyperlane-starknet/src/utils.rs

```diff
@@ -0,0 +1,290 @@
+use std::{future::Future, sync::Arc};
+
+use cainome::cairo_serde::CairoSerde;
+use hyperlane_core::{
+    ChainCommunicationError, ChainResult, HyperlaneMessage, ModuleType, ReorgPeriod,
+};
+use starknet::{
+    accounts::SingleOwnerAccount,
+    core::{
+        chain_id::{MAINNET, SEPOLIA},
+        types::{EmittedEvent, FieldElement, MaybePendingTransactionReceipt},
+        utils::{cairo_short_string_to_felt, CairoShortStringToFeltError},
+    },
+    providers::{jsonrpc::HttpTransport, AnyProvider, JsonRpcClient, Provider, ProviderError},
+    signers::LocalWallet,
+};
+use url::Url;
+
+use crate::{
+    contracts::{
+        interchain_security_module::ModuleType as StarknetModuleType,
+        mailbox::Bytes as MailboxBytes, mailbox::Message,
+        validator_announce::Bytes as ValidatorAnnounceBytes,
+    },
+    HyperlaneStarknetError,
+};
+
+/// Polls a function until it returns true or the max poll count is exceeded.
+pub async fn assert_poll<F, Fut>(f: F, polling_time_ms: u64, max_poll_count: u32)
```

Comment:
there's a similar function to this that we use everywhere else in the hyperlane monorepo, would be better to reuse than one to avoid code duplication. It's called `call_and_retry_n_times`, from `hyperlane-core/src/rpc_clients/retry.ts`

Reply:
replaced